OCPRHV-591: Add oVirt support to the agent side

bkhizgiy opened the pull request:
To add oVirt provider support some modifications need to be done on the agent side.
oVirt platform is detected according to the family parameter and not by product name 
which can be various, to detect the right platform and add it to the isVitual list a new 
function was added and it will change the host product type to match oVirt one, it will
ensure that the rest of the flow will remain the same as for the different types of hosts.

avishayt commented on src/inventory/system_vendor.go:
```diff
@@ -8,7 +8,11 @@ import (
 	"strings"
 )
-
+const (
+	ovirtProductName string = "oVirt"
+	rhvProductName string = "RHV"
+	)
```
Why is this defined here but vmware consts defined in internal/common/platform.go?  Can we keep them consistent?

bkhizgiy replied:
Sure will remove it from here and move it to the same place as VMware, I wasn't aware of the definition in the other file.

bkhizgiy commented:
@avishayt Can you please review the patch? It's a really small change and I think it's ready to be merged.
all staff related to the service side will be merged after the enhancement doc is approved.

avishayt commented on src/inventory/system_vendor.go:
```diff
@@ -8,7 +8,7 @@ import (
 	"strings"
 )
-
+	
```
Remove whitespace

bkhizgiy replied:
done

avishayt commented on src/inventory/system_vendor.go:
```diff
@@ -26,6 +27,11 @@ func isVirtual(product string) bool {
 	return false
 }
 
+// For oVirt VMs the correct platform can be detected only by the Family value, this function is used to check that and update the productName accordingly.
```
Break into 2 lines

bkhizgiy replied:
done

rollandf commented:
/ok-to-test

rollandf commented:
What about adding a test here? src/inventory/system_vendor_test.go

bkhizgiy commented:
> What about adding a test here? src/inventory/system_vendor_test.go

Good Idea, I've added a new test for oVirt product detection and modified the virtual detection one.
